docs/docs/guides/next.mdx

```diff
@@ -6,72 +6,113 @@ sidebar_custom_props:
   docCardIconName: nextjs-dark
 ---
 
-# Next.js
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
 
-In this guide you will learn how to use the
-[hanko-elements](https://github.com/teamhanko/hanko/blob/main/frontend/elements/README.md) web components to
-add authentication and a user profile to your Next.js application.
+# Integrate Hanko with Next.js
 
-## Install dependencies
+Learn how to quickly add authentication and user profile in your Next.js app using Hanko.
 
-Install the `@teamhanko/hanko-elements` package:
+## Install @teamhanko/hanko-elements
+
+Once you've initialized your Next app, installing hanko-elements provides you with access to the prebuilt components: `hanko-auth` and `hanko-profile`.
 
 ```shell npm2yarn
 npm install @teamhanko/hanko-elements
 ```
 
+## Add the Hanko API URL
+
+Retrieve the API URL from the [Hanko console](https://cloud.hanko.io/) and place it in your .env file.
+
+```sh title=".env"
+HANKO_API_URL=https://f4****-4802-49ad-8e0b-3d3****ab32.hanko.io
+```
+
 ## Add `<hanko-auth>` component
 
-To provide a login interface in your app, use the `<hanko-auth>` web component. To do so, first import the `register` function
-from `@teamhanko/hanko-elements` in your Next.js component. Call it with the URL of the Hanko API as an argument to register
-the `<hanko-auth>` element with
-the browser's [`CustomElementRegistry`](https://developer.mozilla.org/de/docs/Web/API/CustomElementRegistry).
-Then use the element in your JSX.
+The `<hanko-auth>` web component adds a login interface to your app. Begin by importing the register function from `@teamhanko/hanko-elements` into your Next.js component. Call it with the Hanko API URL as an argument to register `<hanko-auth>` with the browser's [CustomElementRegistry](https://developer.mozilla.org/de/docs/Web/API/CustomElementRegistry). Once done, include it in your JSX.
 
-:::info
+```mdx-code-block
+<Tabs groupId="environment">
+<TabItem value="app-directory" label="App directory">
+```
 
-If you are using [Hanko Cloud](https://cloud.hanko.io), you can find the API URL on your project dashboard.
-If you are self-hosting you need to provide the URL of your running Hanko backend.
+```jsx title="components/HankoAuth.tsx" showLineNumbers
+"use client";
 
-:::
+import { useEffect } from "react";
+import { register } from "@teamhanko/hanko-elements";
+
+const hankoApi = process.env.HANKO_API_URL;
```

Comment:
If we're rendering this component on client side, I suppose `process.env.HANKO_API_URL` would be null. Since this is next.js app we should add prefix `NEXT_PUBLIC` here.